addons/website/static/src/components/resource_editor/resource_editor.js

```diff
@@ -375,7 +375,15 @@ export class ResourceEditor extends Component {
      */
     async saveXML(resource) {
         const { id, arch } = resource;
-        const context = { ...this.context, lang: false };
+        if (!this.website.currentWebsite.metadata.defaultLang) {
+            // TODO: Remove in master.
```

Comment:
in master Maybe consider a specific route instead of a ir.ui.view.write, that would get the language code in python?
what do you think
That would solve the need of this extra rpc..